kv/memberlist/memberlist_client.go

```diff
@@ -399,6 +404,9 @@ func (m *KV) buildMemberlistConfig() (*memberlist.Config, error) {
 	mlCfg.AdvertiseAddr = m.cfg.AdvertiseAddr
 	mlCfg.AdvertisePort = m.cfg.AdvertisePort
 
+	mlCfg.Label = m.cfg.Label
+	mlCfg.SkipInboundLabelCheck = m.cfg.SkipInboundLabelCheck
```

Comment:
Below we log:

```
level.Info(m.logger).Log("msg", "Using memberlist cluster node name", "name", mlCfg.Name)
```

I would move that log at the end of the function, and change it to something like:

```
level.Info(m.logger).Log("msg", "Using memberlist cluster label %q and node name %q", mlCfg.Label, mlCfg.Name)
```

When manually testing this, I found it useful seeing the actual label.